module/data/action.mjs

```diff
@@ -1026,64 +1038,50 @@ export default class CrucibleAction extends foundry.abstract.DataModel {
   /*  Animation                                   */
   /* -------------------------------------------- */
 
+  /**
+   * Define an animation sequence ID used for this Action
+   * @returns {object}
+   */
+  get _getAnimationPreset() {
+    return {id: undefined}
+  }
+
   /**
    * Prepare a Sequencer animation.
-   * @returns {null}
+   * @returns {null|Promise<Sequencer.Sequence>}
    */
   getAnimationSequence() {
-
-    // Confirm that animation paths exist
-    const config = this._getAnimationConfiguration();
-    if ( !config?.src ) return null;
-    const paths = Sequencer.Database.searchFor(config.src);
-    if ( !paths.length ) return null;
+    const {id, ...options} = this._getAnimationPreset();
+    if ( !Sequencer.Presets.get(id) ) return null;
 
     // Get the origin token
     const originToken = CrucibleAction.#getTargetToken(this.actor);
     if ( !originToken ) return null;
 
-    // Create the Sequence and configure it based on the gesture used
+    // Create the sequence
     const sequence = new Sequence();
-
-    // Single target
-    if ( this.target.type === "single" ) {
-      for ( const outcome of this.outcomes.values() ) {
-        if ( !outcome.rolls.length ) continue;
-        const targetToken = CrucibleAction.#getTargetToken(outcome.target);
-        const hit = outcome.rolls.some(r => r.isSuccess || (r.data.damage?.total > 0));
-        const wait = config.wait ?? 0;
-        if ( config.sequence instanceof Function ) {
-          config.sequence(sequence, config, {originToken, targetToken, hit});
+    switch ( this.target.type ) {
+      case "fan":
+        sequence.effect()
+          .preset(id, {template: this.template, ...options})
+          .play()
+        break;
+      case "single":
+        for ( const outcome of this.outcomes.values() ) {
+          if ( !outcome.rolls.length ) continue;
+          const hit = outcome.rolls.some(r => r.isSuccess || (r.data.damage?.total > 0));
+          const targetToken = CrucibleAction.#getTargetToken(outcome.target);
+          sequence.effect()
+            .preset(id, {source: originToken, target: targetToken, missed: !hit, ...options})
+            .play()
```

Comment:
You may benefit from moving the `.play()` call out of the `for` loop, as each outcome would play the entire assembled Sequence several times, which could duplicate effects that have already been played.